frameworks/keyed/solid/src/main.jsx

```diff
@@ -1,66 +1,70 @@
-    <div class='jumbotron'><div class='row'>
-      <div class='col-md-6'><h1>SolidJS Keyed</h1></div>
-      <div class='col-md-6'><div class='row'>
-        <Button id='run' text='Create 1,000 rows' fn={ run } />
-        <Button id='runlots' text='Create 10,000 rows' fn={ runLots } />
-        <Button id='add' text='Append 1,000 rows' fn={ add } />
-        <Button id='update' text='Update every 10th row' fn={ update } />
-        <Button id='clear' text='Clear' fn={ clear } />
-        <Button id='swaprows' text='Swap Rows' fn={ swapRows } />
-      </div></div>
-    </div></div>
-    <table class='table table-hover table-striped test-data'><tbody>
-      <For each={( state.data )} transform={ selectWhen(() => state.selected, 'danger') }>{ row =>
-        <tr model={ row.id }>
-          <td class='col-md-1' textContent={ row.id } />
-          <td class='col-md-4'><a onClick={ select }>{( row.label )}</a></td>
-          <td class='col-md-1'><a onClick={ remove }><span class='glyphicon glyphicon-remove' aria-hidden='true' /></a></td>
-          <td class='col-md-6'/>
-        </tr>
-      }</For>
-    </tbody></table>
-    <span class='preloadicon glyphicon glyphicon-remove' aria-hidden="true" />
-  </div>
-}
-
-createRoot(() => document.getElementById("main").appendChild(<App />))
+import { createState } from 'solid-js';
+import { render, selectWhen } from 'solid-js/dom';
+
+let idCounter = 1;
+const adjectives = ["pretty", "large", "big", "small", "tall", "short", "long", "handsome", "plain", "quaint", "clean", "elegant", "easy", "angry", "crazy", "helpful", "mushy", "odd", "unsightly", "adorable", "important", "inexpensive", "cheap", "expensive", "fancy"],
+  colours = ["red", "yellow", "blue", "green", "pink", "brown", "purple", "brown", "white", "black", "orange"],
+  nouns = ["table", "chair", "house", "bbq", "desk", "car", "pony", "cookie", "sandwich", "burger", "pizza", "mouse", "keyboard"];
+
+function _random (max) { return Math.round(Math.random() * 1000) % max; };
+
+function buildData(count) {
+  let data = new Array(count);
+  for (let i = 0; i < count; i++) {
+    data[i] = {
+      id: idCounter++,
+      label: `${adjectives[_random(adjectives.length)]} ${colours[_random(colours.length)]} ${nouns[_random(nouns.length)]}`
+    }
+  }
+  return data;
+}
+
+const Button = ({ id, text, fn }) =>
+  <div class='col-sm-6 smallpad'>
+    <button id={ id } class='btn btn-primary btn-block' type='button' onClick={ fn }>{ text }</button>
+  </div>
+
+const App = () => {
+  let rowId;
+  const [ state, setState ] = createState({ data: [], selected: null }),
+    run = () => setState({ data: buildData(1000), selected: null }),
+    runLots = () => setState({ data: buildData(10000), selected: null }),
+    add = () => setState('data', d => [...d, ...buildData(1000)]),
+    update = () => setState('data', { by: 10 }, 'label', l => l + ' !!!'),
+    swapRows = () => setState('data', d => d.length > 998 ? { 1: d[998], 998: d[1] } : d),

```

Comment:
Also, that isn't the kind of work you talking about, that's the complete opposite of "when other libraries actually check what has changed" ...

Reply:
Sort of. This is expressive syntax for the same iterative processes you would be using based on [Netflix's Falcor](https://netflix.github.io/falcor/) and [ImmutableJS](https://immutable-js.github.io/immutable-js/). This still does the for loop to update the data and where applicable runs it through reconciliation. You can see this as a declarative replacement of using Proxy assignment. This way writes require access to the setter in order to update values for control, but under the hood we have mutation.

So is this different than a Virtual DOM library? yes. Could it be used to solely directly proxy DOM changes? yes. In fact a few libraries like [crui](https://github.com/cruijs/crui) are experimenting in this direction to avoid reconciliation where possible. It isn't something I would discount if I could make it work well enough in the general case without increasing the API surface. Obviously the examples I gave were all true, since all the libraries use some sort of list reconciliation, but you are correct that fine grain libraries use events to update data rather than reconciling everything. The only reason the VDOM authors don't cry foul, and some have tried (https://github.com/krausest/js-framework-benchmark/issues/230), is that these are the tests that make the least difference. That and the API surface is still about the data, and not being tied to a single view (even if in this test the data is only tied to a single view).

I was pointing out what I did in this issue, so that you understood how changing swap wouldn't just end the discussion. There is no problem with it being the way it is. I'm kind of glad you didn't change it if you didn't feel it was idiomatic Mikado. Sure I shot down down making a different sort of benchmark here, but my opinion of Mikado is still the same as when I first saw it here https://github.com/krausest/js-framework-benchmark/pull/632#issuecomment-529171129. It isn't the developer experience I prefer even remotely(personal opinion) but I'm not surprised it exists. In fact I anticipated your arrival https://github.com/krausest/js-framework-benchmark/pull/398#issuecomment-392296780 (Read the Edit, if you extend it beyond the specific cheat being discussed). If I seem adversarial it is just that the more combative your tone got the more incentivized I am to correct you. I have no issue with Mikado being how it is.

Reply:
OMG

Reply:
You may simply forget, that I got 3 shit storms in a row like a witch-hunt ... and you was sledging the hammer also.

Reply:
Nobody of yours makes anything better ... if you would like to see a 100% strict implementation, then take a look here: https://github.com/krausest/js-framework-benchmark/pull/656 and learn from it.

Reply:
I know I sound like a broken record at this point. But I will say it again. I have no issue with how Mikado works. Your impression of me slamming down the hammer, is simply because you set yourself up by saying things are just not true or lead me to believe you are missing the point.

Like this statement you just made about your Proxy solution. In this case I haven't had the chance yet to check whether your statement is true, just follow me for a second. Let's pretend I look and based on what I explained in https://github.com/krausest/js-framework-benchmark/pull/655#discussion_r341264466 I realize that in fact this isn't "100% clean implementation(probably the most clean/strict of all)". Because of the absoluteness of your statement that pretend, I know not to be true at all, I respond and drop the hammer so to speak and my response carries weight. You leave feeling dejected and frustrated, and I felt I had no choice but correcting you because the statement was so absolute that I felt it could be misleading. Does it any point mean I dislike what Mikado is doing? No. Do I think that Mikado using Proxies is a super cool feature that could move Web Development forward definitely. I want to see all the potential of Proxies. It would just that I felt the need to clarify. I'm pretty sure there are ways to avoid this in the future. The easiest might simply be conscious of talking in absolutes unless you want to invite longer discussions.

Reply:
Ok sorry, it might we are different. I just need some seconds to look on the code to realize what it is. The implementation of my proxy benchmark is so simple, that I couldn't believe that I need to make any explanations to it.

Reply:
I mean from what you described above I believe the Proxy implementation is as you say. I just was trying to come up with some hypothetical example of what has been happening to this point.